src/assert.js

```diff
@@ -52,7 +52,11 @@ class Assert {
 			acceptCallCount = 1;
 		}
 
-		test.usedAsync = true;
+		// If we have already used async, we don't want to set this
+		if ( !test.usedAsync ) {
+			test.usedAsync = !allowAssertionsAfter;
```

Comment:
Should the `usedAsync` property be renamed, if we are now using it to determine if an error should be thrown for asserting (or resolving async) too late?